### PR TITLE
added support for nested groups

### PR DESCRIPTION
**Pull request by ar4s**

### CODE REVIEW

File: src/ralph/accounts/management/commands/ldap_sync.py [hunk 1]
```diff
@@ -31,15 +33,48 @@ def _truncate(field_key, field_name, ldap_dict):
...
+            settings, 'AUTH_LDAP_NESTED_FILTER', '(memberOf:{})'
+        )
+        self.group_users = {}
+        with LDAPConnectionManager() as conn:
+            for group_name, values in nested_groups.items():
+                ldap_filter = nested_filter.format(values['filter'])
+                users = conn.search_s(
+                    values['base_dn'],
+                    ldap.SCOPE_SUBTREE,
+                    '(&(objectClass=user){})'.format(ldap_filter))
```
Comment (mkurek) [inline src/ralph/accounts/management/commands/ldap_sync.py]: use `settings.LDAP_SERVER_OBJECT_USER_CLASS` here

File: src/ralph/accounts/management/commands/ldap_sync.py [hunk 2]
```diff
@@ -52,19 +87,18 @@ def _create_or_update_user(self, user_dn, ldap_dict):
...
-        ldap_user.populate_user()
+        return ldap_user.populate_user()
 
     def _disconnect(self):
         self.conn.unbind_s()
 
     def _run_ldap_query(self, query):
-        self.conn = get_ldap()
-        try:
+        with LDAPConnectionManager() as conn:
```
Comment (mkurek) [inline src/ralph/accounts/management/commands/ldap_sync.py:121]: :+1:

File: src/ralph/accounts/management/commands/ldap_sync.py [hunk 3]
```diff
@@ -31,15 +33,48 @@ def _truncate(field_key, field_name, ldap_dict):
...
+        nested_groups = getattr(settings, 'AUTH_LDAP_NESTED_GROUPS', {})
+        nested_filter = getattr(
+            settings, 'AUTH_LDAP_NESTED_FILTER', '(memberOf:{})'
+        )
+        self.group_users = {}
+        with LDAPConnectionManager() as conn:
+            for group_name, values in nested_groups.items():
+                ldap_filter = nested_filter.format(values['filter'])
+                users = conn.search_s(
+                    values['base_dn'],
```
Comment (ar4s) [inline src/ralph/accounts/management/commands/ldap_sync.py]: TODO: replace by `AUTH_LDAP_USER_SEARCH_BASE`

**Comment by mkurek** [comment]
:ok:

**Comment by szok** [comment]
:ok:

**Pull request merged**